src/figuresClasses.ts

```diff
@@ -1,19 +1,73 @@
+type Colors = 'red' | 'green' | 'blue';
+type Shapes = 'triangle' | 'circle' | 'rectangle';
+
 export interface Figure {
+  color: Colors,
+  shape: Shapes,
 
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shapes = 'triangle';
 
-}
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c].sort((x, y) => y - x);
 
-export class Circle {
+    if ((sides[0] >= sides[1] + sides[2])
+        || a < 1 || b < 1 || c < 1) {
+      throw new Error('Every side must me more then 0');
+    }
+  }
 
+  getArea(): number {
+    const { a, b, c } = this;
+    const s: number = (a + b + c) / 2;
+
+    return +(Math.sqrt(s * (s - a) * (s - b) * (s - c))).toFixed(2);
```

Comment:
Краще явно приводити до конкретного типу
```suggestion
    return Number((Math.sqrt(s * (s - a) * (s - b) * (s - c))).toFixed(2));
```

Reply:
okay